### PR TITLE
[Code enhancements] Fix various explicitness, laziness, hard codes

### LINKED ISSUE #7997: [Possible enhancement] Redundant check in VirtualMemory.cpp
https://github.com/RPCS3/rpcs3/blob/d451a0b7b76ef2ee001dd62bed847ee6f33ac9ff/Utilities/VirtualMemory.cpp#L155-L160

@elad335 

## Explanation
m_file is already checked for -1 in loop header. So check is redundant inside "if"

## Suggestion 

```suggestion
	if (errno == EMFILE) 
```

### PR DESCRIPTION
**Pull request by sampletext32**
This PR fixes numerous code confusions. 
Fixes #7997

### CODE REVIEW

File: Utilities/File.cpp [hunk 1]
```diff
@@ -44,7 +44,7 @@ static std::unique_ptr<wchar_t[]> to_wchar(const std::string& source)
 	verify("to_wchar" HERE), MultiByteToWideChar(CP_UTF8, 0, source.c_str(), size, buffer.get() + 32768 + (unc ? 8 : 4), size);
 
 	// Canonicalize wide path (replace '/', ".", "..", \\ repetitions, etc)
-	verify("to_wchar" HERE), GetFullPathNameW(buffer.get() + 32768, 32768, buffer.get(), nullptr) - 1 < 32768 - 1;
+	verify("to_wchar" HERE), GetFullPathNameW(buffer.get() + 32768, 32768, buffer.get(), nullptr) < 32768;
```
Comment (elad335) [inline Utilities/File.cpp:47]: it was checking for zero. this isn't math where both substructions would cancel each other's out.

File: Utilities/File.cpp [hunk 2]
```diff
@@ -1514,8 +1514,9 @@ const std::string& fs::get_config_dir()
 #ifdef _WIN32
 		wchar_t buf[32768];
 		constexpr DWORD size = static_cast<DWORD>(std::size(buf));
-		if (GetEnvironmentVariable(L"RPCS3_CONFIG_DIR", buf, size) - 1 >= size - 1 &&
-			GetModuleFileName(NULL, buf, size) - 1 >= size - 1)
+		// Subtract 1 was removed for speedup reason
+		if (GetEnvironmentVariable(L"RPCS3_CONFIG_DIR", buf, size) >= size &&
+			GetModuleFileName(NULL, buf, size) >= size)
```
Comment (elad335) [inline Utilities/File.cpp:1519]: Again, remove this change as it introduces a bug of not checking zero.

**Comment by Nekotekina** [comment]
Doesn't seem like a good idea to use isdigit, because of its pitfalls (UB if outside of *unsigned* char range).

File: rpcs3/rpcs3qt/cheat_manager.cpp [hunk 3]
```diff
@@ -347,16 +347,16 @@ std::vector<u32> cheat_engine::search(const T value, const std::vector<u32>& to_
 	}
 	else
 	{
-		// Looks through mapped memory
-		for (u32 page_ind = (0x10000 / 4096); page_ind < (0xF0000000 / 4096); page_ind++)
+		// Looks through mapped memory (removed division by 4096 for speedup reasons)
```
Comment (elad335) [inline rpcs3/rpcs3qt/cheat_manager.cpp:350]: Avoid commenting things in code where you just want to mention them for the pull request's reviewer.
Just mention them in commit/pull request description.

**Comment by sampletext32** [comment]
> Doesn't seem like a good idea to use isdigit, because of its pitfalls (UB if outside of _unsigned_ char range).

This is done mostly for readability reasons. And btw, it works with std::string[indexed] value which are only basic chars. So it doesn't seem like UB for me.

File: rpcs3/rpcs3qt/cheat_manager.cpp (hunk 3 above)
Comment (sampletext32) [reply rpcs3/rpcs3qt/cheat_manager.cpp:350]: Ok, understand

File: rpcs3/rpcs3qt/cheat_manager.cpp [hunk 4]
```diff
@@ -245,12 +245,12 @@ bool cheat_engine::resolve_script(u32& final_offset, const u32 offset, const std
...
 	while (index < red_script.size())
 	{
-		if (red_script[index] >= '0' && red_script[index] <= '9')
+		if (std::isdigit(red_script[index]))
 		{
 			std::string num_string;
 			for (; index < red_script.size(); index++)
 			{
-				if (red_script[index] < '0' || red_script[index] > '9')
+				if (!std::isdigit(red_script[index]))
```
Comment (elad335) [inline rpcs3/rpcs3qt/cheat_manager.cpp:253]:
```suggestion
				if (!std::isdigit(static_cast<u8>(red_script[index])))
```
This fixes the UB mentioned earlier.

File: rpcs3/rpcs3qt/cheat_manager.cpp [hunk 5]
```diff
@@ -245,12 +245,12 @@ bool cheat_engine::resolve_script(u32& final_offset, const u32 offset, const std
 
 	while (index < red_script.size())
 	{
-		if (red_script[index] >= '0' && red_script[index] <= '9')
+		if (std::isdigit(red_script[index]))
```
Comment (elad335) [inline rpcs3/rpcs3qt/cheat_manager.cpp:248]: same.

**Comment by sampletext32** [comment]
Offtop.
Thanks to this project, I learned how to make forks, pull requests, resolve code reviews and finally force-push. 
Really appreciate your work.

File: rpcs3/rpcs3qt/cheat_manager.cpp [hunk 6]
```diff
@@ -348,15 +348,15 @@ std::vector<u32> cheat_engine::search(const T value, const std::vector<u32>& to_
 	else
 	{
 		// Looks through mapped memory
-		for (u32 page_ind = (0x10000 / 4096); page_ind < (0xF0000000 / 4096); page_ind++)
+		for (u32 page_start = (0x10000); page_start < (0xF0000000); page_start+=4096)
```
Comment (JohnHolmesII) [inline rpcs3/rpcs3qt/cheat_manager.cpp:351]:
```suggestion
		for (u32 page_start = (0x10000); page_start < (0xF0000000); page_start += 4096)
```

**Comment by AniLeo** [comment]
You may want to squash some commits

File: rpcs3/rpcs3qt/cheat_manager.cpp (hunk 6 above)
Comment (Megamouse) [inline rpcs3/rpcs3qt/cheat_manager.cpp:351]:
```suggestion
		for (u32 page_start = (0x10000); page_start < (0xF0000000); page_start += 4096)
```

File: rpcs3/rpcs3qt/cheat_manager.cpp [hunk 7]
```diff
@@ -348,15 +348,15 @@ std::vector<u32> cheat_engine::search(const T value, const std::vector<u32>& to_
...
 		{
-			if (vm::check_addr(page_ind * 4096))
+			if (vm::check_addr(page_start))
 			{
 				// Assumes the values are aligned
 				for (u32 index = 0; index < 4096; index += sizeof(T))
 				{
-					if (*vm::get_super_ptr<T>((page_ind * 4096) + index) == value_swapped)
-						results.push_back((page_ind * 4096) + index);
+					if (*vm::get_super_ptr<T>((page_start) + index) == value_swapped)
```
Comment (Megamouse) [inline rpcs3/rpcs3qt/cheat_manager.cpp:358]:
```suggestion
					if (*vm::get_super_ptr<T>(page_start + index) == value_swapped)
```

File: rpcs3/rpcs3qt/cheat_manager.cpp [hunk 8]
```diff
@@ -348,15 +348,15 @@ std::vector<u32> cheat_engine::search(const T value, const std::vector<u32>& to_
...
+						results.push_back((page_start) + index);
```
Comment (Megamouse) [inline rpcs3/rpcs3qt/cheat_manager.cpp:359]:
```suggestion
						results.push_back(page_start + index);
```

File: rpcs3/Emu/CPU/CPUThread.cpp [hunk 9]
```diff
@@ -359,7 +359,7 @@ void cpu_thread::operator()()
 		std::this_thread::sleep_for(1ms);
 	}
 
-	if (id_type() == 1 && false)
+	if (false && id_type() == 1)
```
Comment (Megamouse) [inline rpcs3/Emu/CPU/CPUThread.cpp:362]: Tbh you can probably use a switch or similar for id_type.
It's redundant to check for id_type 2 below anyway, if the id_type is 1

Comment (sampletext32) [reply rpcs3/Emu/CPU/CPUThread.cpp:362]: Well, actually you are right, but I wasn't correcting any actual logic, especially in CPU area. In this particular area I fixed a lazy evaluatable check. I will look into this.

As for the others, will be done soon.

**Comment by sampletext32** [comment]
> You may want to squash some commits

Done

File: rpcs3/Emu/CPU/CPUThread.cpp [hunk 10]
```diff
@@ -359,14 +359,18 @@ void cpu_thread::operator()()
...
 
-	if (id_type() == 1 && false)
-	{
-		g_fxo->get<cpu_profiler>()->registered.push(id);
-	}
-
-	if (id_type() == 2 && g_cfg.core.spu_prof)
+	switch (id_type())
 	{
+	case 1: break;
```
Comment (Megamouse) [inline rpcs3/Emu/CPU/CPUThread.cpp:364]: Don't do weird stuff like that.
Just comment out the command below

**Pull request merged**